nextflow-parsing/NextflowParsingFunction/src/main/java/io/dockstore/nextflowparsing/App.java

```diff
@@ -0,0 +1,171 @@
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Handler for requests to Lambda function. */
+public class App
+    implements RequestHandler<APIGatewayProxyRequestEvent, APIGatewayProxyResponseEvent> {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(App.class);
+  ObjectMapper mapper = new ObjectMapper();
+
+  /**
+   * Get a language parsing response from NextflowHandler.
+   *
+   * @param nextflowHandler NextflowHandler that contains all relevant Nextflow information
+   * @return LanguageParsingResponse constructed after getting information from NextflowHandler
+   */
+  public static LanguageParsingResponse getResponse(NextflowHandler nextflowHandler) {
+    LanguageParsingResponse response = new LanguageParsingResponse();
+    String descriptorAbsolutePathString = nextflowHandler.getDescriptorTempAbsolutePath();
+    response.setClonedRepositoryAbsolutePath(descriptorAbsolutePathString);
+    VersionTypeValidation versionTypeValidation = nextflowHandler.getVersionTypeValidation();
+    if (nextflowHandler.getDescriptorContents() == null) {
+      versionTypeValidation.setValid(false);
+      Map<String, String> messageMap = new HashMap<>();
+      // TODO: Don't use temp absolute path
+      messageMap.put(nextflowHandler.getDescriptorTempAbsolutePath(), "File not found");
+      versionTypeValidation.setMessage(messageMap);
+    }
+    if (nextflowHandler.getConfiguration() == null) {
+      versionTypeValidation.setValid(false);
+    }
+    // TODO: This should be an array, don't join
+    response.setAuthor(
+        String.join(", ", NextflowUtilities.getAuthors(nextflowHandler.getConfiguration())));
+    response.setDescription(NextflowUtilities.getDescription(nextflowHandler.getConfiguration()));
+    versionTypeValidation.setValid(true);
+    response.setVersionTypeValidation(versionTypeValidation);
+    response.setSecondaryFilePaths(nextflowHandler.getSecondaryDescriptorPaths());
+    return response;
+  }
+
+  @Override
+  public APIGatewayProxyResponseEvent handleRequest(
+      final APIGatewayProxyRequestEvent input, final Context context) {
+
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Content-Type", MediaType.APPLICATION_JSON);
+
+    APIGatewayProxyResponseEvent response = new APIGatewayProxyResponseEvent().withHeaders(headers);
+    if (input != null && input.getBody() != null) {
+      try {
+        LanguageParsingRequest request =
+            mapper.readValue(input.getBody(), LanguageParsingRequest.class);
+        try {
+          String s =
+              parseFile(
+                  request.getUri(),
+                  request.getBranch(),
+                  request.getDescriptorRelativePathInGit(),
+                  request);
```

Comment:
If you're going to send the `LanguageParsingRequest` object, it doesn't seem like you need to send individual properties of the request as well.

Reply:
I don't think I'm sending individual properties of the request?

Reply:
I mean if you change it to `parseFile(request)`, the `parseFile` implementation can invoke `request.getUri()`, etc. itself.